modules/kuberay-monitoring/main.tf

```diff
@@ -12,14 +12,24 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# google managed prometheus engine
+locals {
+  gmp_namespace = var.autopilot_cluster ? "gke-gmp-system" : "gmp-system"
+}
+
+# Temporary workaround to ensure the GMP webhook is installed before applying PodMonitorings.
+# This dependency can be removed once the GMP webhook is moved from node to KCP.
+resource "null_resource" "wait_for_gmp" {
+  provisioner "local-exec" {
+    command = "while ! kubectl rollout status -n ${local.gmp_namespace} deploy/gmp-operator; do sleep 5; done"
```

Comment:
This might be slightly easier to read:
```
kubectl wait --all pods -n gke-gmp-system --for=condition=Ready --timeout=300s
```

Reply:
I actually changed it to look for the presence of a configmap produced by the deployment. This should be a more reliable indicator and reduce flakiness.

Unfortunately, `kubectl wait` only works on created objects.

Reply:
`kubectl watch` also wouldn't work in the case that `gmp-operator` has not yet been applied (race condition). The `null_resource` would never be created in that case.

Reply:
@artemvmin  if #403  works (we tested manually @hsachdevah  and it worked ), then maybe we can just include a dummy deployment that would trigger nodes, and also gmp-operator stables during that.. after that actual application deployment would not show any flakiness.

Reply:
@umeshkumhar I think the concern is that #403 is only fixing the CI tests, but it doesn't actually resolve the flake when actually deploying the terraform module. 

If we take the dummy deployment approach, it needs to be included in every Autopilot deployment, not just in the tests. But personally I would like to avoid dummy deployment if possible

Reply:
I have an idea for a provider-based solution that doesn't use kubectl. I'll work on that today.

Reply:
The provider solution was too involved considering we have a GMP solution coming down the pike. Opting for a simple delay.